api/cleaner.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	log "github.com/sirupsen/logrus"
+)
+
+// cleanerInterval generates the cleaner interval from the baseInterval and a max splay
+func cleanerInterval(baseInterval, maxSplay string) (*time.Duration, error) {
+	base, err := time.ParseDuration(baseInterval)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("parsed base interval (%s) of %f seconds", baseInterval, base.Seconds())
+
+	maxs, err := time.ParseDuration(maxSplay)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("parsed max splay interval (%s) of %f seconds", maxSplay, maxs.Seconds())
+
+	r := rand.Int63n(int64(maxs))
+	interval := base + time.Duration(r)
+	log.Infof("starting cleaner with interval of %fs", interval.Seconds())
+
+	return &interval, nil
+}
+
+// run starts the cleaner and listens for a shutdown call.
+func (c *cleaner) run() {
+	ticker := time.NewTicker(c.interval)
+	go func() {
+		// Loop that runs forever
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
The mixing of cases with abbreviated naming kind of blows my mind:  ticker.C then c.action.

Reply:
I agree that `C` is an unfortunately named function, but its from the `stdlib` 

https://golang.org/pkg/time/#Ticker

Reply:
lolz - whaddayagonnado. It's all so clear to me now - thx for the link.